lib/spack/spack/schema/modules.py

```diff
@@ -127,8 +127,9 @@
                 'type': 'object',
                 'patternProperties': {
                     # prefix-relative path to be inspected for existence
-                    r'\w[\w-]*': {
-                        '$ref': '#/definitions/array_of_strings'}}},
+                    r'\w[\w-]*': array_of_strings
```

Comment:
Aaah. Isn't that better?

Reply:
It's more readable indeed (but the schema can't be moved to a resource file external to Python code, which some time ago I thought it could have happened).